components/BottomNavigation/src/MDCBottomNavigationBar.m

```diff
@@ -0,0 +1,267 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomNavigationBar.h"
+
+#import "MaterialMath.h"
+#import "MaterialRTL.h"
+#import "private/MDCBottomNavigationItemView.h"
+
+static const CGFloat kMDCBottomNavigationBarHeight = 72.f;
+static NSString *const kMDCBottomNavigationBarBadgeColorString = @"badgeColor";
+static NSString *const kMDCBottomNavigationBarBadgeValueString = @"badgeValue";
+static NSString *const kMDCBottomNavigationBarImageString = @"image";
+static NSString *const kMDCBottomNavigationBarTitleString = @"title";
+static NSString *const kMDCBottomNavigationBarNewString = @"new";
+
+@interface MDCBottomNavigationBar ()
+
+@property(nonatomic, strong) NSMutableArray<MDCBottomNavigationItemView *> *itemViews;
+@property(nonatomic, strong) UIView *containerView;
+
+@end
+
+@implementation MDCBottomNavigationBar
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCBottomNavigationBarInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonMDCBottomNavigationBarInit];
+  }
+  return self;
+}
+
+- (void)commonMDCBottomNavigationBarInit {
+  self.backgroundColor = [UIColor whiteColor];
+  self.autoresizingMask = (UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleWidth);
+  _selectedItemTintColor = [UIColor blackColor];
+  _unselectedItemTintColor = [UIColor grayColor];
+  _titleHideState = MDCBottomNavigationBarTitleHideStateDefault;
+  _containerView = [[UIView alloc] initWithFrame:CGRectZero];
+  _containerView.autoresizingMask = (UIViewAutoresizingFlexibleLeftMargin |
+                                     UIViewAutoresizingFlexibleRightMargin);
+  [self addSubview:_containerView];
+  _itemViews = [NSMutableArray array];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  // Accommodate bottom inset for iPhone X.
+  CGFloat bottomInset = 0;
+  if (@available(iOS 11.0, *)) {
```

Comment:
Sadly you need to wrap this in `#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
` to not break older versions of Xcode